tests/index.ts

```diff
@@ -0,0 +1,165 @@
+import * as fs from "fs";
+import * as path from "path";
+import { extractLandingState, ISetupSettings, ITestTarget, TrunkDriver } from "tests/driver";
+import { ILinterVersion, ITestingArguments } from "tests/types";
+
+let printed_args = false;
+
+// Parse the environment variable-specified linter version. This can either be
+// - KnownGoodVersion, which parses the linter definition and attempts to specify a known_good_version
+// - Latest, which automatically retrieves the latest linter version if network connectivity is available
+// - A specified version. Note that this will apply to all tests, so only use this environment variable when tests are filtered
+const parseLinterVersion = (value: string): ILinterVersion | string | undefined => {
+  if (value == "KnownGoodVersion") {
+    return ILinterVersion.KnownGoodVersion;
+  } else if (value == "Latest") {
+    return ILinterVersion.Latest;
+  } else if (value && value.length > 0) {
+    return value;
+  }
+  return undefined;
+};
+
+// Parse the global testing config inputs, specified as environment variables.
+// - PLUGINS_TEST_CLI_VERSION replaces the repo-wide trunk.yaml's specified cli-version
+// - PLUGINS_TEST_CLI_PATH specifies an alternative path to a trunk binary
+// - PLUGINS_TEST_LINTER_VERSION specifies a linter version semantic (see 'parseLinterVersion')
+export const parseInputs = () => {
+  const args = <ITestingArguments>{
+    cliVersion: process.env.PLUGINS_TEST_CLI_VERSION,
+    cliPath: process.env.PLUGINS_TEST_CLI_PATH,
+    linterVersion: parseLinterVersion(process.env.PLUGINS_TEST_LINTER_VERSION ?? ""),
+  };
+  if (!printed_args && (args.cliVersion || args.cliPath || args.linterVersion)) {
+    console.debug(args);
+    printed_args = true;
+  }
+  return args;
+};
```

Comment:
nits: this can be done outside of this function and exported as a static const, also unidiomatic typing of an object

```suggestion
export const ARGS: ITestingArguments = {
  cliVersion: process.env.PLUGINS_TEST_CLI_VERSION,
  cliPath: process.env.PLUGINS_TEST_CLI_PATH,
  linterVersion: parseLinterVersion(process.env.PLUGINS_TEST_LINTER_VERSION ?? ""),
};
if (ARGS.cliVersion || ARGS.cliPath || ARGS.linterVersion) {
  console.debug(ARGS);
}
```